app/src/main/java/io/gnosis/safe/ui/settings/app/passcode/ChangePasscodeViewModel.kt

```diff
@@ -0,0 +1,70 @@
+package io.gnosis.safe.ui.settings.app.passcode
+
+import io.gnosis.data.repositories.CredentialsRepository
+import io.gnosis.data.security.HeimdallEncryptionManager
+import io.gnosis.safe.Tracker
+import io.gnosis.safe.notifications.NotificationRepository
+import io.gnosis.safe.ui.base.AppDispatchers
+import io.gnosis.safe.ui.base.BaseStateViewModel
+import io.gnosis.safe.ui.settings.app.SettingsHandler
+import javax.inject.Inject
+
+
+class ChangePasscodeViewModel
+@Inject constructor(
+    private val credentialsRepository: CredentialsRepository,
+    private val notificationRepository: NotificationRepository,
+    private val encryptionManager: HeimdallEncryptionManager,
+    private val settingsHandler: SettingsHandler,
+    private val tracker: Tracker,
+    appDispatchers: AppDispatchers
+) : BaseStateViewModel<ChangePasscodeViewModel.ChangePasscodeState>(appDispatchers) {
+
+    override fun initialState(): ChangePasscodeState = ChangePasscodeState(viewAction = null)
+
+    fun removeOwner() {
+        safeLaunch {
+            credentialsRepository.owners().forEach {
+                credentialsRepository.removeOwner(it)
+            }
+            // Make sure all owners are deleted at this point
+            if (credentialsRepository.ownerCount() == 0) {
+                updateState { ChangePasscodeState(AllOwnersRemoved) }
+            } else {
+                updateState { ChangePasscodeState(OwnerRemovalFailed) }
```

Comment:
why do we have it and how to react to it? I would rather define OwnerRemovalFailed as Throwable and extend our error handler. Fragment should react in viewAction is ShowError ->